customer-service/src/main/scala/domain/services/CustomerService.scala

```diff
@@ -4,17 +4,20 @@ import cats.MonadThrow
 import cats.data.OptionT
 import cats.syntax.all._
 import domain.repositories.CustomerRepository.CustomerRepository
-import domain.{Customer, CustomerId, CustomerWithDetails}
+import domain.{Customer, CustomerDetails, CustomerId, CustomerWithDetails}
 
 class CustomerService[F[_]: MonadThrow](repository: CustomerRepository[F])(
-    customerDetailsService: CustomerDetailsService[F]
+    customerDetailsServices: CustomerDetailsService[F]
 ) {
   def create(customer: Customer): F[Unit]            = repository.add(customer)
   def createBatch(customers: Set[Customer]): F[Unit] = repository.addBatch(customers)
 
   def get(id: CustomerId): OptionT[F, CustomerWithDetails] = for {
-    customer        <- repository.get(id)
-    customerDetails <- customerDetailsService.getCustomerDetails(id.value)
-  } yield CustomerWithDetails(customer.id, customer.name, customerDetails)
+    customer <- repository.get(id)
+    customerWithDetails <- OptionT.liftF(customerDetailsServices.getCustomerDetails(id.value).value.map {
```

Comment:
ur pattern matching on Option and then recreate Option structure:

      case **some** => CustomerWithDetails(customer.id, customer.name, **some**)
      case **none**  => CustomerWithDetails(customer.id, customer.name, **none**)


probably it's enough to have 
```
customerDetailsServices.getCustomerDetails(id.value).value.map(details => CustomerWithDetails(customer.id, customer.name, details))
```